tests/cbmc/proofs/s2n_array_insert_and_copy/s2n_array_insert_and_copy_harness.c

```diff
@@ -50,10 +50,5 @@ void s2n_array_insert_and_copy_harness()
         if (old_array.len != 0 && index == old_array.len) {
             assert_byte_from_blob_matches(&array->mem, &old_byte);
         }
-        uint32_t old_capacity = old_array.mem.size / old_array.element_size;
-        if (old_array.len >= old_capacity) {
-            uint32_t new_capacity = array->mem.size;
-            assert(array->mem.size == (2 * old_capacity * array->element_size));
-        }
```

Comment:
Why don't you think this is helpful? This is basically mapping a functional property of this operation (the same goes to the others) where it ensures the capacity was updated as intended by
```c
    if (array->len >= current_capacity) {
        /* Enlarge the array */
        uint32_t new_capacity = 0;
        GUARD_AS_RESULT(s2n_mul_overflow(current_capacity, 2, &new_capacity));
        GUARD_RESULT(s2n_array_enlarge(array, new_capacity));
    }
```
Do you think this is not being mapped properly? Do you think there is redundancy here? In proof harnesses, usually there is a bit of redundancy, so we can check the operation is behaving as intended.

Reply:
We talked offline. I'm going to remove the check from all the proofs except the s2n_array_insert one.